analyzers/tests/SonarAnalyzer.TestFramework/Verification/DiagnosticVerifier.cs

```diff
@@ -27,21 +27,20 @@ public static class DiagnosticVerifier
     private const string AD0001 = nameof(AD0001);
     private const string LineContinuationVB12 = "BC36716";  // Visual Basic 12.0 does not support line continuation comments.
 
-    public static void Verify(
+    public static int Verify(
             Compilation compilation,
             DiagnosticAnalyzer analyzer,
             string additionalFilePath = null,
             string[] onlyDiagnostics = null,
             string[] additionalSourceFiles = null) =>
         Verify(compilation, [analyzer], CompilationErrorBehavior.FailTest, additionalFilePath, onlyDiagnostics, additionalSourceFiles);
 
-    public static void Verify(
-            Compilation compilation,
-            DiagnosticAnalyzer[] analyzers,
-            CompilationErrorBehavior checkMode, // ToDo: Remove this parameter in https://github.com/SonarSource/sonar-dotnet/issues/8588
-            string additionalFilePath = null,
-            string[] onlyDiagnostics = null,
-            string[] additionalSourceFiles = null)
+    public static int Verify(Compilation compilation,
```

Comment:
All we need here is an information if any issue was found or not, so the minimal needed data is `bool`

Reply:
I'm concerned about the method's semantics. From just looking at `bool Verify(...)`, you might expect the return value to be the result of the verification. It doesn't matter too much because it's only used in one place anyway, I guess.

Reply:
I tried implementing this and found the following drawbacks in addition to the semantic issue mentioned above:
- The code for aggregating the return values gets a bit more complex.
- The error message of `issuesFound.Should().BeTrue("...")` is a bit harder to understand.
- The number of issues found could be helpful when debugging (if only in scarce circumstances).
All of these are very minor, but I think the benefit of switching to `bool` is also very minor, which is why I keep it this way. Feel free to follow up with me on this if you want to discuss it again.